awx/main/tasks.py

```diff
@@ -1146,6 +1149,36 @@ def run(self, pk, **kwargs):
         '''
         Run the job/task and capture its output.
         '''
+        self.instance = self.model.objects.get(pk=pk)
+        containerized = self.instance.is_containerized
+        pod_manager = None
+        if containerized:
+            # TODO: Document what's going on here
+            from awx.main.scheduler.kubernetes import PodManager # Avoid circular import
+            pod_manager = PodManager(self.instance)
+            self.cleanup_paths.append(pod_manager.kube_config)
+            try:
+                pod_manager.deploy()
+            except (ApiException, Exception) as exc:
+                if isinstance(exc, ApiException) and exc.status == 403:
+                    try:
+                        if 'exceeded quota' in json.loads(exc.body)['message']:
+                            logger.warn(exc.body)
+                            logger.warn(f"Could not launch pod for job {self.instance.log_format}. Exceeded quota.")
+                            self.update_model(pk, status='pending')
+                            schedule_task_manager()
+                            return
+
+                    except:
+                        logger.exception("Invalid response from Kubernetes API")
+
+
+                self.update_model(pk, status='error')  # other k8s failures should set the status to error
+                logger.exception('k8s error happened')
```

Comment:
We don't need this line.

Reply:
Actually, I'm wrong.  We *do* need this line so that non-403 errors are logged (https://github.com/ansible/awx/pull/4922/files#r331726876)

That said, we should improve this error message.